nsgcli/nsggrok_main.py

```diff
@@ -0,0 +1,155 @@
+"""
+This module implements Cmd commands that parses text input with Grok using grok-api service
+
+:copyright: (c) 2018 by Happy Gears, Inc
+:license: Apache2, see LICENSE for more details.
+
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import select
+import json
+from cmd import Cmd
+
+import nsgcli.api
+
+
+class NsgGrokCommandLine(Cmd):
+
+    def __init__(self, base_url=None, token=None, netid=1, pattern=None, timeout_set=180):
+        Cmd.__init__(self)
+        self.base_url = base_url
+        self.token = token
+        self.netid = netid
+        self.timeout_sec = timeout_set
+        self.pattern = pattern
+
+    def do_q(self, _):
+        """Quits the program."""
+        self.do_quit()
+
+    @staticmethod
+    def do_quit():
+        """Quits the program."""
+        print('Quitting.')
+        raise SystemExit
+
+    def is_error(self, response):
+        if isinstance(response, dict) and 'error' in response:
+            error = response.get('error', '')
+            return error
+        if isinstance(response, list):
+            return self.is_error(response[0])
+        if isinstance(response, str):
+            resp_obj = json.loads(response)
+            return self.is_error(resp_obj)
+        return None
+
+    def summary(self):
+        print()
+        print('Base url: {0}'.format(self.base_url))
+        print('To exit, enter "quit" or "q" at the prompt')
+
+    ##########################################################################################
+    def do_log(self, cmd_args):
+        """
+        Parse syslog message with grok patterns
+
+        log <message>
+
+        Parameters:
+            message - Syslog Message to be parsed, read stdin if not specified
+
+        Examples:
+            log "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host '10.1.1.1'"
+            log <<< "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host 10.1.1.1"
+        """
+
+        if cmd_args is None:
+            self.read_stdin("log", self.pattern)
+        else:
+            self.call_grok_api("log", cmd_args, self.pattern)
+
+    ##########################################################################################
+    def do_text(self, cmd_args):
+        """
+        Parse text line with grok patterns.
+        Require 'pattern' parameter.
+
+        --pattern="Hello world of %{WORD:world_name}" text <text>
+
+        Parameters:
+            text - Text to be parsed, read stdin if not specified
+
```

Comment:
it looks like you support multi-line input where each line is parsed separately (but I am 100% sure I read this right). If so, please reflect this in the docstring and script "usage" text

Reply:
added example with stdin/multiline